fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/cursors/MergeCursor.java

```diff
@@ -101,22 +101,22 @@
     @SuppressWarnings("squid:S1452")
     @Nonnull
     static <T, S extends MergeCursorState<T>> CompletableFuture<?> whenAny(@Nonnull List<S> cursorStates) {
-        List<S> nonExhausted = new ArrayList<>(cursorStates.size());
+        List<S> nonDoneCursors = new ArrayList<>(cursorStates.size());
         for (S cursorState : cursorStates) {
-            if (!cursorState.isExhausted()) {
+            if (cursorState.mightHaveNext()) {
                 if (MoreAsyncUtil.isCompletedNormally(cursorState.getOnNextFuture())) {
                     // Short-circuit and return immediately if we find a state that is already done.
                     return AsyncUtil.DONE;
                 } else {
-                    nonExhausted.add(cursorState);
+                    nonDoneCursors.add(cursorState);
```

Comment:
Even given my familiarity with cursor logic, this change is really subtle to me. Can we add a comment above line 111 that clarifies what the possibilities are? For example:
```
// The cursor has not hit a limit and the has next future is either incomplete or completed exceptionally.
// In the latter case, anyOf(nonDoneCursors) will also complete exceptionally if no other future is ready.
```
Even that is a bit confusing, unfortunately. 🤷‍♂️

I suppose the tests are the best check against regressions.